src/hangman.rs

```diff
@@ -0,0 +1,106 @@
+#[derive(Default)]
+pub struct Hangman {
+    word: String,
+    attempts: usize,
+    misses: [char; 5],
+    hits: Vec<char>,
+}
+
+pub enum AttemptResult {
+    CanKeepPlaying,
+    Win,
+    Hanged,
+}
+
+const SYMBOL_LETTER_TO_COMPLETE:char = '_';
+
+impl Hangman {
+    pub fn new(word: String) -> Hangman {
+        let vec_size = word.len();
+        Hangman { 
+            word, 
+            attempts: 0, 
+            misses: Default::default(), 
+            hits: vec![SYMBOL_LETTER_TO_COMPLETE; vec_size],
+        }
+    }
+
+    fn guess_letter(&mut self, l:char) -> AttemptResult {
+        
+        if self.hits.contains(&l) || self.misses.contains(&l) {
+            return AttemptResult::CanKeepPlaying;
+        }
+
+        let mut guessed_letter = false;
+        let mut missing_letters = false;
+
+        for (i, c) in self.word.chars().enumerate(){
```

Comment:
These variables should have a descriptive name. Please change it.

Reply:
Changed c to letter_from_word but left i as the index